src/xml/reader.cr

```diff
@@ -28,6 +28,13 @@ class XML::Reader
     )
   end
 
+  def finalize
+    if reader = @reader
+      LibXML.xmlFreeTextReader(reader)
+      @reader = Pointer(Void).null.as(LibXML::XMLTextReader)
```

Comment:
Is this last line needed? 🤔

Reply:
Trying to avoid errors if `finalize` is called multiple times.

Reply:
Not sure if it's strictly necessary, though. If not, we don't need the condition either.

Reply:
I don't think `finalize` could be called manually.

Reply:
could -> should

Reply:
But it can 🤷

Reply:
But it shouldn't!

Reply:
If you do that, well, face the consequences.

Reply:
Please, either make it impossible to misuse (compiler error if called) or safe to misuse (noop).

Reply:
Okay. Then prevent calling `finalize` and continuing using the object too while at it ;-)

Reply:
Can we merge this as is? It's better to be extra safe than sorry. The discussion about being able to call `finalize` is a different one (I guess #6331).

Reply:
There is no concensus on whether `#finalize` could anticipate multiple calls, even within the same module:

https://github.com/crystal-lang/crystal/blob/1c33a0b1d14a1e2c8835bb1b7381789d648624ea/src/llvm/pass_manager_builder.cr#L38-L40

https://github.com/crystal-lang/crystal/blob/1c33a0b1d14a1e2c8835bb1b7381789d648624ea/src/llvm/pass_builder_options.cr#L22-L27

The latter was added in #12116 ;-)